codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/CodegenUtils.java

```diff
@@ -369,4 +370,46 @@ public int compare(MemberShape a, MemberShape b) {
                     .compareTo(symbolProvider.toMemberName(b));
         }
     }
+
+    /**
+     * Attempts to find the first member by exact name in the containing structure. If the member is not found an
+     * exception will be thrown.
+     *
+     * @param shape structure containing member
+     * @param name  member name
+     * @return MemberShape if found
+     */
+    public static MemberShape expectMember(StructureShape shape, String name) {
+        return expectMember(shape, name::equals);
+    }
+
+    /**
+     * Attempts to find the first member by name using a member name predicate in the containing structure. If the
+     * member is not found an exception will be thrown.
+     *
+     * @param shape               structure containing member
+     * @param memberNamePredicate member name to search for
+     * @return MemberShape if found
+     */
+    public static MemberShape expectMember(StructureShape shape, Predicate<String> memberNamePredicate) {
+        return expectMemberByPredicate(shape, (p) -> memberNamePredicate.test(p.getMemberName()));
+    }
+
+    /**
+     * Attempts to find the first member by a predicate in the containing structure. If the member is not found an
+     * exception will be thrown.
+     *
+     * @param shape                structure containing member
+     * @param memberShapePredicate member shape to search for
+     * @return MemberShape if found
+     */
+    public static MemberShape expectMemberByPredicate(
+            StructureShape shape,
+            Predicate<MemberShape> memberShapePredicate
+    ) {
+        return shape.getAllMembers().values().stream()
+                .filter(memberShapePredicate)
+                .findFirst()
+                .orElseThrow(() -> new CodegenException("did not find member in structure shape, " + shape.getId()));
+    }
```

Comment:
These can be merged.

Reply:
Sure, i'd initially had them separate so that a search member by predicate could be available, but it can always be added in later if needed.